### DISCUSSION TITLE
Unarchiving a privat thread channel and adding a new member fails

### DISCUSSION THREAD

**Discussion by Ascend1984** [discord-jda/JDA#2533] [Questions and Help] [answered]
I'm intending to work with threads in my setup and want them to unarchive if i need to perform actions on them. So i implemented the unarchiving with 
`ThreadChannelManager manager = threadChannels.get(i).getManager().setArchived(false).complete();`
But when trying to add a new member to the thread i get `java.lang.IllegalStateException: Cannot modify a ThreadChannel while it is archived!`

Looking into discord i see the thread being active again. So what would the correct way be to unarchive the thread? I get the feeling this is some chaching problem?! Or are there any scenarios where my intended process just will not work?

Thanks in advance!

**Comment by MinnDevelopment** [comment, marked as answer]
You can try something like this:

```java
channel.getManager().setArchived(false)
  .delay(Duration.ofSeconds(1)) // 1 second later
  .map(v -> jda.getThreadChannelById(channel.getIdLong())) // get fresh instance from cache
  .flatMap(thread -> thread.addThreadMember(user)) // then add a member
  .queue(); // important not to use complete() here so cache can catch up!
```

The issue here is definitely the cache, archived threads are never cached by JDA, so the state of that object is never updated. Besides that, discord also has some race-conditions related to archiving, so it is best to wait a little before adding a new member.

**Reply by Ascend1984** [reply]
I needed to rewrite my approach on when I do this, but your suggestion did work. 

I wanted to keep the unarchiving separate from the adding of the member, but could not get it to work. Still always got the cache problem telling me that the thread was still archived.
